cabal-install/Distribution/Client/Setup.hs

```diff
@@ -1366,6 +1368,13 @@ installOptions showOrParseArgs =
           installOnlyDeps (\v flags -> flags { installOnlyDeps = v })
           (yesNoOpt showOrParseArgs)
 
+      , option [] ["index-snapshot"]
+          "Operate on package index state rolled back to given unix-timestamp"
```

Comment:
"repository-snapshot"
"Use the state of the package repositories as of a given timestamp e.g. 2016..."  <- add an example or format string